test/Fields/reduction_cuda.jl

```diff
@@ -115,3 +130,106 @@ end
+        "running reduction test on $device device; h_elem = $h_elem; z_elem = $z_elem; npoly = $npoly; R = $R; z_max = $z_max; FT = $FT",
+    )
+    # horizontal space
+    domain = Domains.SphereDomain(R)
+    horizontal_mesh = Meshes.EquiangularCubedSphere(domain, h_elem)
+    horizontal_topology = Topologies.Topology2D(
+        context,
+        horizontal_mesh,
+        Topologies.spacefillingcurve(horizontal_mesh),
+    )
+    horizontal_topology_cpu = Topologies.Topology2D(
+        context_cpu,
+        horizontal_mesh,
+        Topologies.spacefillingcurve(horizontal_mesh),
+    )
+    quad = Spaces.Quadratures.GLL{npoly + 1}()
+    h_space = Spaces.SpectralElementSpace2D(horizontal_topology, quad)
+    h_space_cpu = Spaces.SpectralElementSpace2D(horizontal_topology_cpu, quad)
+
+    # vertical space
+    z_domain = Domains.IntervalDomain(
+        Geometry.ZPoint(zero(z_max)),
+        Geometry.ZPoint(z_max);
+        boundary_tags = (:bottom, :top),
+    )
+    z_mesh = Meshes.IntervalMesh(z_domain, nelems = z_elem)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    z_topology_cpu = Topologies.IntervalTopology(context_cpu, z_mesh)
+
+    z_center_space = Spaces.CenterFiniteDifferenceSpace(z_topology)
+    z_center_space_cpu = Spaces.CenterFiniteDifferenceSpace(z_topology_cpu)
+
+    z_face_space = Spaces.FaceFiniteDifferenceSpace(z_topology)
+    z_face_space_cpu = Spaces.FaceFiniteDifferenceSpace(z_topology_cpu)
+
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(h_space, z_center_space)
+    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
+
+    hv_center_space_cpu =
+        Spaces.ExtrudedFiniteDifferenceSpace(h_space_cpu, z_center_space_cpu)
+    hv_face_space_cpu =
+        Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space_cpu)
+
+    Yc = set_initial_condition(hv_center_space)
+    Yf = set_initial_condition(hv_face_space)
+
+    Yc_cpu = set_initial_condition(hv_center_space_cpu)
+    Yf_cpu = set_initial_condition(hv_face_space_cpu)
+
+    resultc = Base.sum(Yc)[]
+    resultc_cpu = Base.sum(Yc_cpu)
+
+    resultf = Base.sum(Yf)[]
+    resultf_cpu = Base.sum(Yf_cpu)
+
+    @test resultc_cpu ≈ resultc
+    @test resultf_cpu ≈ resultf
+
+    @test resultc ≈ (4 / 3) * π * ((R + z_max)^3 - R^3) rtol = 1e-2
+    @test resultf ≈ (4 / 3) * π * ((R + z_max)^3 - R^3) rtol = 1e-2
+
+
+    Yc = set_simple_field(hv_center_space)
+    Yc_cpu = set_simple_field(hv_center_space_cpu)
+
+    @test Base.maximum(identity, Yc)[] ≈ Base.maximum(identity, Yc_cpu)
+    @test Base.minimum(identity, Yc)[] ≈ Base.minimum(identity, Yc_cpu)
+
+    @test Statistics.mean(Yc)[] ≈ Statistics.mean(Yc_cpu)
+
+    @test LinearAlgebra.norm(Yc, 1)[] ≈ LinearAlgebra.norm(Yc_cpu, 1)
+    @test LinearAlgebra.norm(Yc, 2)[] ≈ LinearAlgebra.norm(Yc_cpu, 2)
+    @test LinearAlgebra.norm(Yc, 3)[] ≈ LinearAlgebra.norm(Yc_cpu, 3)
+    @test LinearAlgebra.norm(Yc, Inf)[] ≈ LinearAlgebra.norm(Yc_cpu, Inf)
```

Comment:
can these return a scalar?

Reply:
It can currently handle mutiple fields. Should we specialize this for just `Nf = 1`?

Reply:
Modified to return a scalar (result for the first field)